core/src/main/java/hudson/util/FormApply.java

```diff
@@ -87,4 +93,22 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object nod
             }
         };
     }
+
+    /**
+     * Generates the response for the asynchronous background form submission (AKA the Apply button),
+     * that will show a notification of certain type and with provided message. Supported notification types
+     * are {@code SUCCESS}, {@code WARNING} and {@code ERROR}.
```

Comment:
Probably a good idea to reference the UI documentation that specifies the notification types.

Reply:
I guess the best we can do is link to https://github.com/jenkinsci/jenkins/blob/74610e024a6b8fd8feccdc51b8f7741aa6c30e3b/war/src/main/js/components/notifications/index.js#L13-L25 Would that be good enough?

Reply:
Awkward that there's no documentation for this, so probably the best we can do.